libsofia-sip-ua/tport/ws.c

```diff
@@ -805,7 +805,7 @@ ssize_t ws_close(wsh_t *wsh, int16_t reason)
 		wsh->uri = NULL;
 	}
 
-	if (reason && wsh->sock != ws_sock_invalid) {
+	if (wsh->logical_established && reason && wsh->sock != ws_sock_invalid) {
```

Comment:
Avoid calling "ws_raw_write" before "SSL_accept" succeeds  ， Calling "ws_raw_write" before "SSL_accept" succeeds may cause blocking risk。

Reply:
On the other hand when `logical_established` is set `handshake` is also set.

Reply:
If logical_established is not set, there should be no need to send WSOC_CLOSE.

I have studied this problem for a long time, and the real reason is that the socket does not have FIONBIO for unknown reasons


The modifications I submit can be made without blocking when the socket does not have FIONBIO

@seven1240   My English is not clear, I have talked to serven about this problem


int ws_handshake(wsh_t *wsh)
{
....
	*(wsh->buffer + wsh->datalen) = '\0';
}




ssize_t ws_raw_read(wsh_t *wsh, void *data, size_t bytes, int block)
{
...
	if (r > 0) {
		*((char *)data + r) = '\0';
	}
}


Causing the socket to have no FIONBIO flag, it may be due to the memory exceeding the bounds in these two locations, and more testing is needed to determine